gubernator/static/pr.js

```diff
@@ -0,0 +1,82 @@
+
+var upArrow = "\u25b2";
+var downArrow = "\u25bc";
+
+function sortRows(rows, col, colName, dir) {
+  // customize sorting functions for different columns
+  var extract = function(el) { return el.innerText.replace(/^\s+|\s+$/g, ''); }
+  var fn = function(x) { return x; }
+  switch (colName) {
+  case 'Number': fn = Number.parseInt; break;
+  case 'Size': fn = function(x) { return ['XS', 'S', 'M', 'L', 'XL', 'XXL'].indexOf(x); }; break;
+  case 'Updated': extract = function(el) { return +el.firstChild.dataset.epoch; }; break;
+  case 'Author': case 'Assignees': fn = function(x) { return x.toLowerCase(); }; break;
+  }
+
+  for (var i = 0; i < rows.length; i++) {
+    rows[i].index = i;
+  }
+
+  rows.sort(function(a, b) {
+    var valA = fn(extract(a.children[col]));
+    var valB = fn(extract(b.children[col]));
+    if (valA === valB) {
+      return a.index - b.index;  // make the sort stable
+    } else if (valA < valB) {
+      return -dir;
+    } else {
+      return dir;
+    }
+  });
+}
+
+function sort(target) {
+  if (target.nodeName !== "TH") return;
+
+  var reverse = target.innerText[0] == upArrow;
+
+  var col, i = 0;
+
+  for (var sibling of target.parentElement.children) {
+    var first = sibling.innerText[0];
+    if (first == upArrow || first == downArrow) {
+      sibling.innerText = sibling.innerText.slice(2);
+    }
+    if (sibling === target) {
+      col = i;
+    }
+    i++;
+  }
+
+  var table = target.closest('table');
+  var tbody = table.tBodies[0];
+  var rows = Array.prototype.slice.call(tbody.children);
+  tbody.remove();         // clear old rows
+  tbody = table.createTBody();  // create a new body
+
+  sortRows(rows, col, target.innerText, reverse ? -1 : 1);
+
+  for (var row of rows) {
+    tbody.appendChild(row);
+  }
+
+  // Add arrow.
+  target.innerText = (reverse ? downArrow : upArrow) + ' ' + target.innerText;
+
+  return true;
+}
+
+// Handle sorting
+function clickHandler(evt) {
+  var target = evt.target;
+  if (sort(target)) {
+    evt.preventDefault();
+    return true;
+  }
+  return false;
+}
+
+window.addEventListener('load', function() {
+  console.log('hi');
+  document.addEventListener('click', clickHandler, false);  
```

Comment:
As @fejta  mentioned about things related to the TH tag, this could be avoided by either: 

1. Add the EventListener to the table, this would narrow the scope of TH elements.
Or
2. Fetch all TH elements of the table and attach EventListeners to them directly. Comparison functions can be passed to the handler using closures.
`allThEls.forEach((el) => { el.addEventListener('click', () => return clickHandler(fn_getter(extract(header_value)));})`

Reply:
Done.

Reply:
I see the event listener attached to the document still. It should be `th.addEventListener(...)` and you also need to change the sort function.